go/cmd/deploy/main.go

```diff
@@ -1,19 +1,411 @@
+	cli.String("commit", "Git commit SHA"),
+	cli.String("registry", "Container registry",
+		cli.Default(DefaultRegistry),
+		cli.EnvVar(EnvRegistry)),
+	cli.Bool("skip-push", "Skip pushing to registry (for local testing)"),
+	cli.Bool("verbose", "Show detailed output for build and deployment operations"),
+	// Control plane flags (internal)
+	cli.String("control-plane-url", "Control plane URL", cli.Default(DefaultControlPlaneURL)),
+	cli.String("auth-token", "Control plane auth token", cli.Default(DefaultAuthToken)),
+}
+
+// Cmd defines the deploy CLI command (renamed from Command to match your existing pattern)
 var Cmd = &cli.Command{
 	Name:  "deploy",
-	Usage: "Deploy a new version of your API (alias for 'version create')",
-	Description: `Deploy is a convenience command that creates a new version of your API.
-	
-This is equivalent to running 'unkey version create'.`,
-
-	// Copy all flags from version create command
-	Flags:  version.Cmd.Commands[0].Flags,
-	Action: version.Cmd.Commands[0].Action,
+	Usage: "Deploy a new version",
+	Description: `Build and deploy a new version of your application.
+Builds a container image from the specified context and
+deploys it to the Unkey platform.
+
+The deploy command will automatically load configuration from unkey.json
+in the current directory or specified config directory.
+
+EXAMPLES:
+    # Deploy using config file (./unkey.json)
+    unkey deploy
+
+    # Deploy with config from specific directory
+    unkey deploy --config=./test-docker
+
+    # Deploy overriding workspace from config
+    unkey deploy --workspace-id=ws_different
+
+    # Deploy with specific context (overrides config)
+    unkey deploy --context=./demo_api
+
+    # Deploy with your own registry
+    unkey deploy \
+      --workspace-id=ws_4QgQsKsKfdm3nGeC \
+      --project-id=proj_9aiaks2dzl6mcywnxjf \
+      --registry=docker.io/mycompany/myapp
+
+    # Local development (skip push)
+    unkey deploy --skip-push
+
+    # Deploy pre-built image
+    unkey deploy --docker-image=ghcr.io/user/app:v1.0.0
+
+    # Show detailed build and deployment output
+    unkey deploy --verbose
+
+If no config file exists, you can create one with:
+    unkey init`,
+	Flags:  DeployFlags,
+	Action: DeployAction,
+}
+
+func DeployAction(ctx context.Context, cmd *cli.Command) error {
+	// Load configuration file
+	configPath, err := config.GetConfigPath(cmd.String("config"))
+	if err != nil {
+		return err
+	}
+
+	cfg, err := config.LoadConfig(configPath)
+	if err != nil {
+		return fmt.Errorf("failed to load config: %w", err)
+	}
+
+	// Merge config with command flags (flags take precedence)
+	finalConfig := cfg.MergeWithFlags(
+		cmd.String("workspace-id"),
+		cmd.String("project-id"),
+		cmd.String("context"),
+	)
+
+	// Validate that we have required fields
+	if err := finalConfig.Validate(); err != nil {
+		return err // Clean error message already
+	}
+
+	opts := &DeployOptions{
```

Comment:
why is it a pointer?

Reply:
Lemme remove get rid of that pointer. It was required to be a pointer in my initial draft of cli take, but its not required now